codegen/src/test/resources/software/amazon/awssdk/codegen/poet/client/test-endpoint-discovery-async.java

```diff
@@ -264,6 +288,10 @@ public final String serviceName() {
             return executeFuture;
         } catch (Throwable t) {
             return CompletableFutureUtils.failedFuture(t);
+        } finally {
+            Optional<MetricPublisher> metricPublisher = MetricUtils.resolvePublisher(clientConfiguration,
+                    testDiscoveryRequiredRequest);
+            metricPublisher.ifPresent(p -> p.publish(apiCallMetricCollector.collect()));
```

Comment:
Publishing needs to happen after execution so we would need to add a completion stage to the future instead of publishing here

Reply:
oh yeah, I guess I just blindly copied the sync change 😅